OpenSim/Simulation/SimbodyEngine/BallJoint.h

```diff
@@ -49,38 +49,22 @@ of position (\f$\vec{u} \neq \dot{\vec{q}}\f$).
 */
 
 class OSIMSIMULATION_API BallJoint : public Joint {
-OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
+    OpenSim_DECLARE_CONCRETE_OBJECT(BallJoint, Joint);
 
-private:
-    static const int _numMobilities = 3;
-//=============================================================================
-// DATA
-//=============================================================================
+    /** Specify the Coordinates of the BallJoint */
+    CoordinateIndex rx{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex ry{ constructCoordinate(Coordinate::MotionType::Rotational) };
+    CoordinateIndex rz{ constructCoordinate(Coordinate::MotionType::Rotational) };
```

Comment:
Chris pointed out yesterday that side-effect initializers like this have an impact on backwards-compatibility of copy construction (not saying its a problem here; it just seems like a good time to mention it). A default copy constructor is fine since it won't invoke the initializers. A fully-implemented copy constructor will work since the author will have said what to do with each member. But a copy constructor that only mentions some of the members would have expected default construction of the unmentioned members. Now instead those members will execute the initialization code, possibly resulting in unwanted side effects.